### ISSUE TITLE
Auto add documentation with possible type variants for Java/C#

### ISSUE THREAD

**Issue by Mingun** [kaitai-io/kaitai_struct#671]
In #670 you can notice, that with different case types in `switch-on` node resulting field type can be reduced to very generic type (for example, `Object` in Java). In order to improve user experience, we can add additional information to field documentation, that will contain all types, that field can contain. For specified example, Javadoc can be supplemented with text:
```java
/**
 * …Javadoc from `doc` attribute...
 *
 * <p>Field can have values of following classes:
 * <ul>
 * <li>{@link Inner}</li>
 * <li>{@code byte[]}</li>
 * </ul>
 */
```

Such addition do not required for languages, that can express returned type in their types. For now that will be useful at least for Java and C#

**Comment by KOLANICH** [comment]
I wonder if java has union types...

**Comment by Mingun** [comment]
No. Without value classes it is useless, because all is reference and you always can store reference in `Object` variable.

**Comment by GreyCat** [comment]
Makes sense. On a minor note, I'm somewhat worried to generate human language content. If we go that route, likely we'll need some proper i18n, templates in many languages, etc...

**Comment by KOLANICH** [comment]
IMHO i18n is not useful at all here. Comments and docstrings in any lang other than English are considered an antipattern. And English is known by any software developer.

**Comment by dgelessus** [comment]
> And English is known by any software developer.

<sup>[[réf. nécessaire]](https://fr.wikipedia.org/wiki/Aide:R%C3%A9f%C3%A9rence_n%C3%A9cessaire)</sup>

More seriously: I can see why we'd want to avoid automatically adding English text here, because currently all text in the generated doc comments comes directly from the ksy file's `doc` and `doc-ref` attributes, which could theoretically be written in any language. Realistically though, users who don't understand written English won't be able to use KS very well anyway, because all the documentation and error messages is in English. A single fixed sentence is also unlikely to be a problem even if you don't understand English well - you'll recognize that the same text is generated every time, so you only need to look up once what the sentence means.
